### PR TITLE
fix(DataTable): update sort order to go ASC -> DESC -> NONE

### PR DESCRIPTION
**Pull request by joshblack**
Updates sort order and tooltip to follow NONE -> ASC -> DESC -> NONE order.

#### Changelog

**New**

**Changed**

* Update to sorting state and tools under Data Table
* Update to Table Header to better reflect logic for next tooltip

**Removed**

### CODE REVIEW

**Review by alisonjoseph — APPROVED** [review]
Thanks for updating this!

File: src/components/DataTable/TableHeader.js [hunk 1]
```diff
@@ -12,11 +12,18 @@ const translationKeys = {
 const translateWithId = (key, { sortDirection, isSortHeader, sortStates }) => {
   if (key === translationKeys.iconDescription) {
     if (isSortHeader) {
-      const order =
-        sortDirection === sortStates.DESC ? 'descending' : 'ascending';
-      return `Sort rows by this header in ${order} order`;
+      // When transitioning, we know that the sequence of states is as follows:
+      // NONE -> ASC -> DESC -> NONE
+      if (sortDirection === sortStates.NONE) {
+        return 'Rort rows by this header in ascending order';
```
Comment (austinbruch) [inline src/components/DataTable/TableHeader.js]: @joshblack Typo `Rort` vs `Sort`?
```suggestion
        return 'Sort rows by this header in ascending order';
```

File: src/components/DataTable/TableHeader.js [hunk 2]
```diff
@@ -12,11 +12,18 @@ const translationKeys = {
...
+      }
+      if (sortDirection === sortStates.ASC) {
+        return 'Rort rows by this header in descending order';
```
Comment (austinbruch) [inline src/components/DataTable/TableHeader.js]: @joshblack Same as above...
```suggestion
        return 'Sort rows by this header in descending order';
```

**Comment by joshblack** [comment]
Thanks @austinbruch!

**Review by lovemecomputer — CHANGES_REQUESTED** [review]
looks good! tho i think there's one CSS change that could make the UX of the component more clear:

in the gif, note that when returned to default sort state, the :focus state remains, and judging by the icon, it makes it look like it has been returned to ASC. that ASC icon (and focus border) remains until clicking outside of the element:
sorting01

**Comment by joshblack** [comment]
@lovemecomputer agreed! Added this issue: https://github.com/IBM/carbon-components/issues/1421

**Review by lovemecomputer — APPROVED** [review]

**Comment by lovemecomputer** [comment]
@joshblack cool! since this is captured in that issue and the fix may come in a separate PR, i'm approving these changes here~

**Comment by joshblack** [comment]
Real quick for anyone looking to merge, let me add `aria-sort` support as well in this PR!

**Comment by joshblack** [comment]
Updated!

**Comment by reinisg** [comment]
Hello!
Sorry for old conversation update, but I have question about this. Is it possible to set one column as default ASC sorted? Is there any parameters which pass to table column? @joshblack 

Regards, Reinis

**Pull request merged**